src/conversions/unitconvert.cpp

```diff
@@ -476,6 +478,8 @@ PJ *CONVERSION(unitconvert,0) {
         }
         if (f != 0.0)
             Q->xy_factor *= f;
+        if (strcmp(normalized_name, "Radian") == 0)
```

Comment:
in case a numeric value is passed, normalized_name will be nullptr, so this should be tested
Same below.
Strange that it doesn't crash on tests

Reply:
> Strange that it doesn't crash on tests

Actually, it does on Linux & Mac :-) Windows must have some nullptr protection in its strcmp() implementation

Reply:
Yes, of course... Is there a better way to do this using C++ functions?

Reply:
An alternative might be to initialize normalized_name = "" and make get_unit_conversion_factor() return "" instead of nullptr

As far as C++ is concerned, you can compare std::string with the natural == operator. But there are caveats: constructing a std::string from a const char* requires for the later to be non-null. And in theory std::string construction from const char* can fail with a out-of-memory C++ exception that should be caught somewhere. c_api.cpp normally makes sure to wrap in try / catch all calls to C++ code that can throw exceptions, but using C++ from "traditional" PROJ code that can be directly called from C exported symbols has the risk of having uncaugh exceptions (I'd note that this is the theory. In practice GDAL for example makes great use of std::string and doesn't bother catching out-of-memory exceptions when the strings are known to be of small size. so there's a theoretical risk of crash in case of low-mem situation).
Furthermore starting to use C++ & exceptions in code that makes manual memory allocations (to be opposed to C++ objects like std::vector that manage their memory) expose to memory leaks if you fail to free objects when exceptions are thrown.

Reply:
I've just added a check for `nullptr`, that is probably the simplest way to go. 

Okay, many good reasons for not mixing in C++ here. Thanks for the explanation!